File: javascript/smartlist/smartlist-api-mysql-project/src/modules/question/dto/response/answer-option.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsInt, IsNotEmpty } from 'class-validator';
import { AnswerOption } from 'src/entities/answer-option.entity';

export class AnswerOptionRespDto {
  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  @IsInt()
  id: number;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  text: string;

  @ApiProperty({
    type: Boolean,
  })
  @IsNotEmpty()
  correct: boolean;

  @ApiProperty({
    type: Number,
  })
  @IsInt()
  score: number;

  constructor(answerOption: AnswerOption, forAssessment: boolean) {
    this.id = answerOption.id;
    this.text = answerOption.text;
    this.score = answerOption.score;
    this.correct = forAssessment ? null : answerOption.correct;
  }
}

File: java/smartlist/src/main/java/com/smartlist/module/question/dto/response/AnswerOptionRespDTO.java
package com.smartlist.module.question.dto.response;

import com.smartlist.model.AnswerOption;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AnswerOptionRespDTO {

    @Schema(description = "Answer option Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer id;

    @Schema(description = "Answer option text", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    String text;

    @Schema(description = "Is correct answer option", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    Boolean correct;

    @Schema(description = "Answer option score", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer score;

    /*
    AnswerOptionRespDTO(AnswerOption answerOption, boolean forAssessment) {
        this.id = answerOption.getId();
        this.text = answerOption.getText();
        this.score = answerOption.getScore();
        this.correct = forAssessment ? null : answerOption.isCorrect();
    }*/


}
